src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDesignTimeDescriptorFactory.cs

```diff
@@ -102,18 +119,10 @@ private XmlDocumentationProvider GetXmlDocumentationProvider(Assembly assembly)
 
             var documentationProvider = _documentationProviderCache.GetOrAdd(cacheKey, valueFactory: _ =>
             {
-                var assemblyLocation = assembly.Location;
-
-                if (string.IsNullOrEmpty(assemblyLocation) && !string.IsNullOrEmpty(assembly.CodeBase))
-                {
-                    var uri = new UriBuilder(assembly.CodeBase);
-
-                    // Normalize the path to UNC path. This will remove things like file:// from start of uri.Path.
-                    assemblyLocation = Uri.UnescapeDataString(uri.Path);
-                }
+                var assemblyLocation = GetAssemblyLocation(assembly);
 
                 // Couldn't resolve a valid assemblyLocation.
-                if (string.IsNullOrEmpty(assemblyLocation))
+                if (assemblyLocation == null)
```

Comment:
`string.Empty` can no longer occur at this point. The diff is misleading, See line 105. The `GetAssemblyLocation` now does the logic that you're referring to.